### PR TITLE
ENH: nagfor - get_flags_linker_so() on darwin

### PR DESCRIPTION
**Pull request by dimpase**
fix NAG's nagfor Fortran compiler (linking) on darwin - was omitted in #19968.

nagfor on Darwin needs the same linker flags as nag, so this simplifies the code, in fact.

### CODE REVIEW

File: numpy/distutils/fcompiler/nag.py [hunk 1]
```diff
@@ -63,7 +60,6 @@ class NAGFORCompiler(BaseNAGFCompiler):
         'archiver'     : ["ar", "-cr"],
         'ranlib'       : ["ranlib"]
         }
-
```
Comment (rgommers) [inline numpy/distutils/fcompiler/nag.py:66]: This is a bit odd, having one blank line between methods is preferred.

**Review by rgommers — COMMENTED** [review]
Thanks @dimpase. This looks reasonable, if it fixes the SciPy issue we should just go ahead and merge it. It cannot break anything else.

File: numpy/distutils/fcompiler/nag.py (hunk 1 above)
Comment (dimpase) [reply numpy/distutils/fcompiler/nag.py]: I've just put the spacing in.

**Comment by charris** [comment]
Thanks @dimpase .

**Comment by dimpase** [comment]
Thanks!

**Comment by ThemosTsikas** [comment]
The doucumented `nagfor` option is `-unsharedrts`, not `-unsharedf95`.

**Comment by rgommers** [comment]
> The doucumented `nagfor` option is `-unsharedrts`, not `-unsharedf95`.

@dimpase do you want to make this change?

**Comment by dimpase** [comment]
yes, sure, will do.

**Pull request merged**